projects/vdk-plugins/vdk-trino/tests/test_vdk_trino_lineage.py

```diff
@@ -27,26 +31,133 @@
     },
 )
 def test_lineage():
-    mock_lineage = mock.MagicMock(LineageLogger)
+
+    mock_lineage_logger = mock.MagicMock(LineageLogger)
 
     class TestConfigPlugin:
         @hookimpl
         def vdk_initialize(self, context):
-            context.state.set(LINEAGE_LOGGER_KEY, mock_lineage)
+            context.state.set(LINEAGE_LOGGER_KEY, mock_lineage_logger)
 
     runner = CliEntryBasedTestRunner(TestConfigPlugin(), trino_plugin)
 
-    result: Result = runner.invoke(["trino-query", "--query", "SELECT 1"])
+    result: Result = runner.invoke(
+        ["trino-query", "--query", f"create table {TEST_TABLE_1} (test_column varchar)"]
+    )
+    cli_assert_equal(0, result)
+    result: Result = runner.invoke(
+        ["trino-query", "--query", f"create table {TEST_TABLE_2} (test_column varchar)"]
+    )
+    cli_assert_equal(0, result)
+
+    #   Test insert query
+    #   -----------------
+
+    insert_query = f"insert into {TEST_TABLE_1} values('test value')"
+    result: Result = runner.invoke(["trino-query", "--query", insert_query])
+    cli_assert_equal(0, result)
+
+    # the lineage data is different on every run of the test,
+    # so we need this class to generalize the dict which is to be matched
+    class InsertLineageDataMatch:
+        def __eq__(self, lineage_data):
+            import json
+
+            print(json.dumps(lineage_data, indent=4))
+            return (
+                lineage_data.keys()
+                >= {"@type", "query", "@id", "status", "outputTable"}
+                and lineage_data["@type"] == "insert_select"
+                and lineage_data["status"] == "OK"
+                and lineage_data["query"] == insert_query
+                and re.fullmatch(r"[0-9]{10}\.[0-9]+", lineage_data["@id"])
+                and lineage_data["outputTable"]["schemaTable"]["table"] == TEST_TABLE_1
+            )
+
+    mock_lineage_logger.send.assert_called_with(InsertLineageDataMatch())
+
+    #   Test select query
```

Comment:
why not split it into multiple test methods ? Having one long test method is generally not a best idea since 
A) it's harder to read (having long methods is not best practice, it's good to have some rule keep methods below N lines)
B) it's less parallelizable
C) it takes more time to troubleshoot (e.g if multiple test fails you discover them one by one.

Reply:
I thought about that and it's cleaner. Although, each test spawns a new Trino container, as I understand, so it will make the execution a lot heavier (although more parallelizable). Do you think I should split?

Reply:
The trino fixture is "session" scope so it should be spawned only once per test session . So it should not be a problem.

Reply:
Great. I will split them.